### PR TITLE
enhance: remove jindoruntime's fsGroup

### PR DESCRIPTION
**Pull request by TrafalgarZZZ**
### Ⅰ. Describe what this PR does
1. Remove any `fsGroup` related transformation in Jindofsx and JindoCache engine. `fsGroup` has no critical effect on JindoRuntime's functionality but it will chown mounted persistent volumes which may cause unexpected changes on file ownership and permissions. See [Configure volume permission and ownership change policy for Pods](https://kubernetes.io/docs/tasks/configure-pod-container/security-context/#configure-volume-permission-and-ownership-change-policy-for-pods) for more information: 

> By default, Kubernetes recursively changes ownership and permissions for the contents of each volume to match the fsGroup specified in a Pod's securityContext when that volume is mounted. For large volumes, checking and changing ownership and permissions can take a lot of time, slowing Pod startup.

2. mount ufs volumes in read-only mode unless the Dataset is explicitly set to `ReadWriteMany`

### Ⅱ. Does this pull request fix one issue?

fixes #XXXX

### Ⅴ. Special notes for reviews

### CODE REVIEW

File: pkg/ddc/jindocache/transform.go [hunk 1]
```diff
@@ -302,6 +301,15 @@ func (e *JindoCacheEngine) transformMaster(runtime *datav1alpha1.JindoRuntime, m
 			if len(value.UFSVolumes) == 0 {
 				value.UFSVolumes = []UFSVolume{}
 			}
+
+			// Default to mount ufs volumes in read-only mode. Mount in read-write mode only when
+			// the dataset is set to ReadWriteMany explicitly.
+			ufsVolumeReadOnly := true
+			accessModes := dataset.Spec.AccessModes
+			if len(accessModes) == 1 && accessModes[0] == corev1.ReadWriteMany {
```
Comment (cheyang) [inline pkg/ddc/jindocache/transform.go:309]: How bout ReadWriteOnce?

**Review by cheyang — CHANGES_REQUESTED** [review]

File: charts/jindofsx/templates/fuse/daemonset.yaml [hunk 2]
```diff
@@ -64,7 +64,6 @@ spec:
       securityContext:
```
Comment (cheyang) [inline charts/jindofsx/templates/fuse/daemonset.yaml:64]: Please also update chart version and changelog. thanks.

File: pkg/ddc/jindocache/transform.go (hunk 1 above)
Comment (TrafalgarZZZ) [reply pkg/ddc/jindocache/transform.go:309]: ReadWriteOnce is now not supported for Fluid datasets.

File: charts/jindofsx/templates/fuse/daemonset.yaml (hunk 2 above)
Comment (TrafalgarZZZ) [reply charts/jindofsx/templates/fuse/daemonset.yaml:64]: Done

**Review by cheyang — APPROVED** [review]
/lgtm
/approve

**Pull request merged**